### PR TITLE
Address and remove all occurrences of the Java annotation @UpdateForV9 (ES-9378)

### PR DESCRIPTION
**Pull request by alexey-ivanov-es**
Removes all occurrences of UpdateForV9 annotation owned by Core/Infra that are not blocked

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java [hunk 1]
```diff
@@ -120,13 +119,7 @@ public static class Names {
     public static final String THREAD_POOL_METRIC_NAME_REJECTED = ".threads.rejected.total";
 
     public enum ThreadPoolType {
-        @Deprecated(forRemoval = true)
-        @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA) // no longer used, remove in v9
-        DIRECT("direct"),
         FIXED("fixed"),
-        @Deprecated(forRemoval = true)
-        @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA) // no longer used, remove in v9
-        FIXED_AUTO_QUEUE_SIZE("fixed_auto_queue_size"),
```
Comment (alexey-ivanov-es) [inline server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:129]: PR for 8.x: https://github.com/elastic/elasticsearch/pull/118492

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java [hunk 2]
```diff
@@ -243,8 +236,6 @@ static NavigableMap<Integer, IndexVersion> getAllVersionIds(Class<?> cls) {
         return Collections.unmodifiableNavigableMap(builder);
     }
 
-    @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA)
-    // We can simplify this once we've removed all references to index versions earlier than MINIMUM_COMPATIBLE
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/index/IndexVersions.java:247]: We shouldn't remove old index versions here - this is something the search team need to take a look at

File: server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java [hunk 3]
```diff
@@ -80,22 +79,20 @@ public void testEqualsHashcodeSerialization() {
...
-        final NodeMetadata nodeMetadata = NodeMetadata.FORMAT.loadLatestState(logger, xContentRegistry(), tempDir);
-        assertThat(nodeMetadata.nodeId(), equalTo("y6VUVMSaStO4Tz-B5BxcOw"));
-        assertThat(nodeMetadata.nodeVersion(), equalTo(BuildVersion.fromVersionId(0)));
+
+        ElasticsearchException ex = assertThrows(
+            ElasticsearchException.class,
+            () -> NodeMetadata.FORMAT.loadLatestState(logger, xContentRegistry(), tempDir)
+        );
+        Throwable rootCause = ex.getRootCause();
+        assertTrue(rootCause instanceof IllegalStateException);
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java:94]: `assertThat(rootCause, instanceOf(IllegalStateException.class))`

File: server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java [hunk 4]
```diff
@@ -80,22 +79,20 @@ public void testEqualsHashcodeSerialization() {
...
+        assertEquals("Node version is required in node metadata", rootCause.getMessage());
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java:95]: `assertThat(rootCause.getMessage(), equalTo(...))`

File: server/src/test/java/org/elasticsearch/index/IndexVersionTests.java [hunk 5]
```diff
@@ -151,8 +151,6 @@ public void testMax() {
         }
     }
 
-    @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA)
-    @AwaitsFix(bugUrl = "believe this fails because index version has not yet been bumped to 9.0")
     public void testMinimumCompatibleVersion() {
         assertThat(IndexVersion.getMinimumCompatibleIndexVersion(7170099), equalTo(IndexVersion.fromId(6000099)));
         assertThat(IndexVersion.getMinimumCompatibleIndexVersion(8000099), equalTo(IndexVersion.fromId(7000099)));
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/index/IndexVersionTests.java:156]: We should have a new assert for the v9 index version

File: server/src/main/java/org/elasticsearch/env/NodeEnvironment.java [hunk 6]
```diff
@@ -1505,28 +1506,17 @@ private static void tryWriteTempFile(Path path) throws IOException {
...
-                }
-            } catch (NumberFormatException e) {
-                // continue and return default
-            }
+    static BuildVersion getBestDowngradeVersion(BuildVersion previousNodeVersion) {
+        if (previousNodeVersion.onOrAfterMinimumCompatible()) {
+            return previousNodeVersion;
         }
-        return "7.17.0";
+        return BuildVersion.fromVersionId(Version.CURRENT.minimumCompatibilityVersion().id);
```
[the remark this answers is not in the document]
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/env/NodeEnvironment.java:1516]: It's not ideal. Maybe we need a `BuildVersion.getMinimumCompatible` method to calculate this without invoking `Version`. @ryan what do you think?

File: server/src/test/java/org/elasticsearch/index/IndexVersionTests.java (hunk 5 above)
Comment (alexey-ivanov-es) [reply server/src/test/java/org/elasticsearch/index/IndexVersionTests.java:156]: Added

File: server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java (hunk 4 above)
Comment (alexey-ivanov-es) [reply server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java]: Done

File: server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java (hunk 3 above)
Comment (alexey-ivanov-es) [reply server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java]: Done

File: test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java [hunk 7]
```diff
@@ -92,21 +92,9 @@ public void testFailure(Failure failure) throws Exception {
         GradleMessageBuilder gradleMessageBuilder = new GradleMessageBuilder(b);
         gradleMessageBuilder.appendAllOpts(failure.getDescription());
 
-        if (isRestApiCompatibilityTest()) {
```
Comment (mark-vieira) [inline test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java:95]: Why are we removing this? Rest compatibility tests are still relevant to 9.0.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

File: test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java (hunk 7 above)
Comment (alexey-ivanov-es) [reply test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java:95]: It seems `tests.restCompat` property is used in the tests only to select nodes for V7 compatibility tests (https://github.com/elastic/elasticsearch/pull/117992/files#diff-8563be9d0ca3fe7e4456eb5712dd42588425972b9848913734f5e177f9b25fa8L333), and this mode is no longer supported in V9

Comment (mark-vieira) [reply test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java:95]: Rest compatibility mode is still a thing in V9, it's just that we only support back to V8, and we still run these tests ([for example](https://gradle-enterprise.elastic.co/s/3ip4zgdx5ua2m)). This infra needs to remain.

Comment (alexey-ivanov-es) [reply test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java]: Sorry, I missed the case where this branch might still be relevant. I've reverted the changes

**Review by mark-vieira — APPROVED** [review]

**Comment by alexey-ivanov-es** [comment]
PR split to:
https://github.com/elastic/elasticsearch/pull/119088
https://github.com/elastic/elasticsearch/pull/119095
https://github.com/elastic/elasticsearch/pull/119101
https://github.com/elastic/elasticsearch/pull/119105

**Pull request closed without merging**